### ISSUE TITLE
[DocDB] Table partition boundaries structure mismatch for indexes

### ISSUE THREAD

**Issue by arybochkin** [yugabyte/yugabyte-db#12191]
Jira Link: [DB-1836](https://yugabyte.atlassian.net/browse/DB-1836)
### Description

Index creation supports `split at` statement to explicitly specify partitions for tablets.
The syntax example:
```
create unique index idx1 on test1(name ASC) split at values(('p'));
```

Taking into account the index table schema has at least one internal hidden column (refer to [documentation](https://github.com/yugabyte/yugabyte-db/blob/c2deaf4af152b08e009f945ca7d662d325d6c46a/src/yb/common/pg_system_attr.h#L38)), it is impossible to specify a value for that particular column inside split at instruction.

Let's assume the following example
```
# create table test1(id int primary key, name text);
# create unique index idx1 on test1(name ASC) split at values(('p'));
```
As internal value cannot be specified, the partition bound looks like `DocKey([], ["p"])`, so the last column from the primary/doc key is completely ignored! The encoded value of the partition bound looks like `"Sp\x00\x00!" == 5370000021`. But the expected partition key should look like `"Sp\x00\x00$!" == 537000002421`, where '$' byte(s) stand for the value of internal index columns (possible values are `kNullLow`, `kNullHigh` or `base table ctid`).

The partition bound structure mismatch may lead to a case when last part of `YBOperation.partition_key` (equals to `0x00`) for the subsequent insert operation is compared against `kGroupEnd == 0x21` and the operation is forwarded to an incorrect tablet. Refer to #12190 for the details.

Once #12190 is fixed, there should be no visible side effects because of the `partition_key` structure mismatch for split at statement, but it is still a good point to fix the issue.

For any questions, contact @arybochkin

**Comment by arybochkin** [comment]
It's not looking as a tablet splitting issue, it's just for `split at` statement, but it's a good point to fix the issue together with #12190 and #12211 in order to cover index recreation within single execution within #12212

**Comment by arybochkin** [comment]
keeping opened to not forget to backport after version mismatch issue is fixed.

**Comment by rthallamko3** [comment]
@arybochkin, This issue is already backported to 2.12 and looks like the fix made it to 2.14 too right? Can we close this now?

**Comment by arybochkin** [comment]
@rthallamko3 there's a single commit for both #12190 and $12191, it's going to be back ported to 2.14 only.

**Comment by arybochkin** [comment]
Fixed and backported to 2.14 in context of #12190, hence closing.
